app/src/main/java/io/apicurio/registry/util/ArtifactTypeUtil.java

```diff
@@ -84,20 +113,47 @@ public static ArtifactType discoverType(ContentHandle content, String contentTyp
         } catch (Exception e) {
             // Apparently it's not JSON.
         }
-        
+
         // Try protobuf (only if we haven't already)
         if (!triedProto) {
             ArtifactType type = tryProto(content);
             if (type != null) {
                 return type;
             }
         }
-        
+
         // Try GraphQL (SDL)
         if (tryGraphQL(content)) {
             return ArtifactType.GRAPHQL;
         }
-        
+
+        // Try the various XML formatted types
+        try (InputStream stream = content.stream()) {
+            Document xmlDocument = threadLocaldocBuilder.get().parse(stream);
+            Element root = xmlDocument.getDocumentElement();
+            String ns = root.getNamespaceURI();
+
+            if (ns == null) {
+                // XML
+                return ArtifactType.XML;
+            }
+
+            // XSD
+            if (ns.contains("www.w3.org/2001/XMLSchema")) {
+                return ArtifactType.XSD;
+            }
+
+            // WSDL
+            if (ns.contains("schemas.xmlsoap.org/wsdl")) {
+                return ArtifactType.WSDL;
+            }
```

Comment:
We should look for specific namespace(s) here rather than use `contains`.  I think we need to check both of these:

* `http://schemas.xmlsoap.org/wsdl/`
* `http://www.w3.org/ns/wsdl`

I do not believe that 2nd namespace ever got much traction, but that is WSDL 2.0's namespace.